specification/msi/resource-manager/Microsoft.ManagedIdentity/2015-08-31-preview/ManagedIdentity.json

```diff
@@ -0,0 +1,535 @@
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{resourceName}": {
+      "put": {
+        "operationId": "UserAssignedIdentities_CreateOrUpdate",
+        "description": "Create or update an identity in the specified subscription and resource group.",
+        "x-ms-examples": {
+          "IdentityCreate": {
+            "$ref": "./examples/IdentityCreate.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/IdentityCreateOrUpdateParameters"
+            },
+            "description": "Parameters to create or update the identity"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Updated identity",
+            "schema": {
+              "$ref": "#/definitions/Identity"
+            }
+          },
+          "201": {
+            "description": "Created identity",
+            "schema": {
+              "$ref": "#/definitions/Identity"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "patch": {
+        "operationId": "UserAssignedIdentities_Update",
+        "description": "Update an identity in the specified subscription and resource group.",
+        "x-ms-examples": {
+          "IdentityUpdate": {
+            "$ref": "./examples/IdentityUpdate.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/IdentityCreateOrUpdateParameters"
```

Comment:
Why not use the "Identity" definition so it can do the roundtrip? After all, the updatable parameters are the only ones not marked as readonly on "Identity"

Reply:
Identity definition has other parameters that are not required in the identity create parameters, like id, name, tenantId, principald, clientId, clientSecretUrl etc.

Reply:
If the only difference is that some of the properties are server generated and immutable, then the correct way to describe them is to use the readOnly attribute. If this is the only difference between the request and response body schemas, this allows you to use the same model/schema definition.

Reply:
Ah. I see..

Reply:
Fixed. Using "Identity" definition in UserAssignedIdentities_CreateOrUpdate Operation